skcosmo/preprocessing/flexible_scaler.py

```diff
@@ -175,13 +178,12 @@ def transform(self, K, y=None):
         :param y: ignored
 
         :return: tranformed matrix Kc
-        """
-        parameters = self.get_params()
-        """check each of the parameters self.reference_shape, self.ref_mean, 
+
+        check each of the parameters self.reference_shape, self.scale, self.ref_mean,
         and self.ref_cmean, which must all be defined
         """
-        for key in parameters.keys():
-            if parameters[key] is None:
+        for key in ["ref_cmean", "ref_mean", "scale", "reference_shape"]:
+            if not hasattr(self, key) or getattr(self, key) is None:
```

Comment:
I do not agree with this listing at all. First, such code makes it difficult for any kind of code extensibility (if needed). Secondly, it overloads the code, and it is absolutely unclear why.

Reply:
So the previous instantiation had a few flaws -- if there were additional class attributes, then the loop would cycle through them and check. Additionally, if none of these were initialized, then this test would totally miss the error. It needs to be explicit.